packages/lumx-react/src/components/dialog/Dialog.test.tsx

```diff
@@ -0,0 +1,145 @@
+import { ESCAPE_KEY_CODE } from '@lumx/core/js/constants';
+import { CLASSNAME, Dialog, DialogProps } from '@lumx/react/components/dialog/Dialog';
+
+import { ICommonSetup, Wrapper, commonTestsSuite } from '@lumx/react/testing/utils';
+
+import { mount, shallow } from 'enzyme';
+import 'jest-enzyme';
+import noop from 'lodash/noop';
+import React from 'react';
+
+import * as stories from './Dialog.stories';
+
+// Mock out the useIntersectionObserver hook since it can't work with Jest/Enzyme.
+jest.mock('@lumx/react/hooks/useIntersectionObserver', () => ({
+    useIntersectionObserver: () => [noop, []],
+}));
+
+/////////////////////////////
+
+/**
+ * Define the overriding properties waited by the `setup` function.
+ */
+type ISetupProps = Partial<DialogProps>;
+
+/**
+ * Defines what the `setup` function will return.
+ */
+interface ISetup extends ICommonSetup {
+    props: ISetupProps;
+
+    /**
+     * The <div> element that wraps the dialog and children elements.
+     */
+    wrapper: Wrapper;
+}
+
+/////////////////////////////
+
+/**
+ * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
+ *
+ * @param props  The props to use to override the default props of the component.
+ * @param     [shallowRendering=true] Indicates if we want to do a shallow or a full rendering.
+ * @return      An object with the props, the component wrapper and some shortcut to some element inside of the
+ *                       component.
+ */
+const setup = ({ ...props }: ISetupProps = {}, shallowRendering = true): ISetup => {
+    const renderer = shallowRendering ? shallow : mount;
+    // @ts-ignore
+    const wrapper = renderer(<Dialog isOpen {...props} />);
+
+    return {
+        props,
+        wrapper,
+    };
+};
+
+describe(`<${Dialog.displayName}>`, () => {
+    // 1. Test render via snapshot.
+    describe('Snapshots and structure', () => {
+        // Do snapshot render test on every stories.
+        for (const [storyName, Story] of Object.entries(stories)) {
+            if (typeof Story !== 'function') {
+                continue;
+            }
+
+            it(`should render story ${storyName}`, () => {
+                // @ts-ignore
+                const wrapper = shallow(<Story />);
+                expect(wrapper).toMatchSnapshot();
+            });
+        }
+    });
```

Comment:
@juanigalan91 @pidupuis One way to do snapshot testing on stories without any storybook or Jest addons

Reply:
ah nice :) definitely a great way to avoid repeating props on both stories and tests !